Cleanup PO files

glogiotatidis opened the pull request.

glogiotatidis commented:
@jgmize r?

jgmize commented on scripts/cleanup_po.py:
```diff
@@ -0,0 +1,59 @@
+    for version, data in settings.VERSIONS_LOCALE_MAP.items():
+        if locale in data.get('locales', []):
+            versions.append(version)
+    return versions
```
I think we also need to handle the case of new versions for a locale that haven't been fully translated or approved for production, and so aren't in settings.VERSIONS_LOCAL_MAP yet.

glogiotatidis replied:
Good point. I added `pending_locales` 
https://github.com/mozilla/masterfirefoxos/pull/115/files#diff-826dbf0a1c425031f6332ca9e1e7fa96R26

glogiotatidis replied:
And we probably need to populate `pending_locales` too, is there a list?

jgmize commented on scripts/cleanup_po.py:
```diff
@@ -0,0 +1,60 @@
+        if (locale in data.get('locales', [])
+            or locale in data.get('pending_locales', [])):
+            versions.append(version)
+    return versions
```
It's outside the scope of this PR, but we might consider changing the VERSIONS_LOCALE_MAP datastructure to be locale centric instead of version centric later.

jgmize replied:
Actually, the docs that you wrote for this PR (thanks, btw) highlight the fact that activating new versions requires editing files and can't be done from the admin, which we will likely want to change anyway (but is even further outside the scope of this PR, sorry).

jgmize commented:
Testing this locally against a freshly updated checkout of the [l10n repo](https://github.com/mozilla/masterfirefoxos-l10n), `./manage.py runscript cleanup_po` wiped out almost all of the existing translations in the po files without preserving the versions that were mapped in the settings.

glogiotatidis commented:
Apologies I had some stashed changes and I forgot to commit. It should work fine now.

jgmize commented:
Sorry @glogiotatidis testing the new version locally I'm still seeing a bunch of strings get stripped out from the po file that need to remain. For example, in the 'de' locale, I'm seeing a bunch of strings that match the 1-1 slug but are still getting stripped out of the po file.

jgmize commented:
Also, we should add 'xx' to the 'pending' locales for all versions to validate that all strings are kept.

jgmize commented:
To make this more concrete, I pushed the results to a branch so I could link to a comparison view: https://github.com/jgmize/masterfirefoxos-l10n/compare/pr-115?diff=split&expand=1&name=pr-115

glogiotatidis commented:
I cannot reproduce your claim @jgmize and the comparison view doesn't really help me. One thing to note is the `po.save()` will re-arrange things, so git diff isn't really useful.

What I did to verify my code is the following:
 - Loaded  db backup from prod
 - Deleted versions 1-4 and 1-3T
 - `./manage.py runscript db_strings` and `./manage.py makemessages -a`
 - cp locales/de/LC_MESSAGES/django.po django.de.orig.po
 - Added a `print()` before line 56 https://github.com/mozilla/masterfirefoxos/pull/115/files#diff-826dbf0a1c425031f6332ca9e1e7fa96R56
 - `./manage.py runscript cleanup_po`
 - No `prints`, therefore no entries removed
 - To exclude the reordering caused by `po.save` I made it conditional. https://github.com/glogiotatidis/masterfirefoxos/commit/7b8bf78700a5e0b626ab29d5e2d8d888f2e5d5d8#diff-826dbf0a1c425031f6332ca9e1e7fa96R60

jgmize commented:
Really sorry about that @glogiotatidis, I can't replicate my earlier issues either; must have been a PEBKAC error. This LGTM, thanks for your patience :)
